src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbGetItemQueryPlanFactory.java

```diff
@@ -0,0 +1,114 @@
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeVisitor;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+import com.exasol.adapter.dynamodb.dynamodbmetadata.DynamodbTableMetadata;
+import com.exasol.adapter.dynamodb.remotetablequery.*;
+
+/**
+ * This class builds a {@link DynamodbGetItemQueryPlan} if possible.
+ */
+public class DynamodbGetItemQueryPlanFactory {
+
+    /**
+     * Builds a {@link DynamodbGetItemQueryPlan} is possible for the given query.
+     * 
+     * @param documentQuery query to build the plan for
+     * @param tableMetadata DynamoDB table metadata used for checking the primary key
+     * @return the generated plan
+     * @throws PlanDoesNotFitException if this query can't be executed using a {@link DynamodbGetItemQueryPlan}
+     */
+    public DynamodbGetItemQueryPlan buildGetItemPlanIfPossible(
+            final RemoteTableQuery<DynamodbNodeVisitor> documentQuery, final DynamodbTableMetadata tableMetadata)
+            throws PlanDoesNotFitException {
+        final Visitor visitor = new Visitor(tableMetadata);
+        try {
+            documentQuery.getSelection().accept(visitor);
+            final Map<String, AttributeValue> key = visitor.getPrimaryKey();
+            checkIfKeyIsComplete(key, tableMetadata);
+            return new DynamodbGetItemQueryPlan(documentQuery.getFromTable().getRemoteName(), key);
+        } catch (final PlanDoesNotFitExceptionWrapper wrapper) {
+            throw wrapper.getWrappedException();
+        }
+    }
+
+    private void checkIfKeyIsComplete(final Map<String, AttributeValue> key, final DynamodbTableMetadata tableMetadata)
+            throws PlanDoesNotFitException {
+        if (key.size() < 1) {
+            throw new PlanDoesNotFitException(
+                    "Not a GetItem request as the partition key was not specified in the where clause.");
+        }
+        if (tableMetadata.getPrimaryKey().hasSortKey() && key.size() != 2) {
+            throw new PlanDoesNotFitException(
+                    "Not a GetItem request as the sort key was not specified in the where clause.");
+        }
+    }
+
+    private static class Visitor implements QueryPredicateVisitor<DynamodbNodeVisitor> {
+        private final DynamodbTableMetadata tableMetadata;
+        private final Map<String, AttributeValue> primaryKey = new HashMap<>();
+
+        private Visitor(final DynamodbTableMetadata tableMetadata) {
+            this.tableMetadata = tableMetadata;
+        }
+
+        @Override
+        public void visit(
+                final ColumnLiteralComparisonPredicate<DynamodbNodeVisitor> columnLiteralComparisonPredicate) {
+            final DocumentPathExpression columnPath = columnLiteralComparisonPredicate.getColumn()
+                    .getPathToSourceProperty();
+            if (columnPath.size() != 1) {
+                return; // This is not an key attribute as it
```

Comment:
The comment seems to be incomplete, or there is a typo. Did you meant, `This is not a key attribute as is.` or should there be rest, `as it does ...`?